analytics_framework/rag_system/setup_chroma_vector_db.py

```diff
@@ -58,18 +59,49 @@ def chromadb_processing(documents, ids, db_collection_name, what_to_query, resul
     return outcome
 
 
-if __name__ == '__main__':
-    documents, ids = yt_data_read_process("../intake/data/YoutubeCommentsDataSet.csv",
-                                          1000,
-                                          "Comment"
-                                          )
-    results = chromadb_processing(documents,
-                                  ids,
-                                  "YT_comments_new",
-                                  "My macbook broke in an accident",
-                                  )
-
-    # voila mix multiple query types to arrive at the desired outcome
-    print(f"The outcome for the query text \n {results}")
-    # just the 'documents' part is interesting to us, let's scoop it out
-    print(results["documents"])
+@click.command()
+@click.option(
+    "--query",
+    "-q",
+    default="My macbook broke in an accident",
+    show_default=True,
+    help="Query text to search for in the ChromaDB collection.",
+)
+@click.option(
+    "--n-results",
+    "-n",
+    "n_results",
+    default=3,
+    show_default=True,
+    type=int,
+    help="Number of nearest neighbours to return from ChromaDB.",
+)
+def main(query: str, n_results: int) -> None:
+    """
+    Populate a ChromaDB collection from YouTube comments and run a query.
+
+    If no query is provided via CLI, a default example query is used.
+    """
```

Comment:
Thanks for the additional thoughts on the docstrings. 
Format below is recommended and unifies across the framework

```python
def function_name(arg1: type, arg2: type) -> return_type:
    """
    Brief description of what the function does.

    Args:
        arg1 (type): Description of the argument.
        arg2 (type): Description of the argument.

    Returns:
        return_type: Description of what is returned.
    """
```